build.sbt

```diff
@@ -593,10 +593,9 @@ lazy val playJson = libraryProject("play-json")
     description := "Provides Play json codecs for http4s",
     startYear := Some(2018),
     libraryDependencies ++= Seq(
-      Http4sPlugin.playJson.cross(CrossVersion.for3Use2_13)
+      Http4sPlugin.playJson
     ),
-    publish / skip := tlIsScala3.value,
-    compile / skip := tlIsScala3.value,
+    crossScalaVersions := Seq(scala_212, scala_213),
```

Comment:
I don't think this works.  They're still aggregated projects, but:
* on `++3.0.2`, these modules are still on 2.13.8
* on `++3.0.2!`, these modules fail on `update`

Reply:
I guess I don't know what "works" means in this case. In my experience this setup is sufficient to get us through CI and publishing and not have issues locally. I use it in feral which is a mix of 2.13/3 for core modules and 2.12 for the sbt plugin.

Reply:
Jawn is also currently using this setup to skip Scala 3 for native:
https://github.com/typelevel/jawn/blob/e6e53c2e19387a4a20cb582a30c16f464e46a348/build.sbt#L36

Reply:
https://github.com/http4s/http4s/runs/4873655533?check_suite_focus=true#step:9:5279

I don't see how to do this and still be able to run a variant of `++3.0.2` to compile Scala 3 locally, unless you're specifically compiling certain modules.  It leaves parts of the aggregate in a bad state.  Would love to learn a better way.

(The better way is to banish these modules to their own repos, but we don't have that infrastructure yet.)

Reply:
Oh, yikes, missed that. Yep, this won't work here :(

Basically what we need is https://github.com/sbt/sbt-projectmatrix. We can do a poor-man's approximation by creating two aggregate roots, one with everything and one with Scala 2 only stuff. Thanks to the changes in sbt-gh-actions we can then add those two roots to the CI matrix and exclude them appropriately on Scala version.

Reply:
Putting aside that complication, what we can do now is what you suggested above:

> If that doesn't work, we could also apply that modifier conditionally for Scala 3.

But then, I think we also need to disable the unused dependencies check for Scala 3, right?

Reply:
We had something like that when Scala 2.13 was new, and it was a pain in the ass running the right root locally.  I don't know if that's still an issue in modern tooling.

I think we could alternatively solve this problem by just applying the `for3Use2_13` on Scala 3 only.  Or by opting out of these tests on these modules until the bug is fixed or the Great Schism.  I don't want to sludge up the whole build for scarcely used modules.

Reply:
> But then, I think we also need to disable the unused dependencies check for Scala 3, right?

That wouldn't be a problem, because they're not published for Scala 3 anyway.